modules/renter/hostdb/hostentry.go

```diff
@@ -113,19 +115,43 @@ func (hdb *HostDB) AverageContractPrice() types.Currency {
 	return totalPrice.Div64(uint64(len(hosts)))
 }
 
-// IsOffline reports whether a host is offline. If the HostDB has no record of
-// the host, IsOffline will return false.
-//
-// TODO: Is this behavior that makes sense?
+// IsOffline reports whether h is offline. A host is considered offline if it
+// has been scanned at least once in the last three days, but hasn't responded
+// to the scan during that period. If the host has not been scanned in the last
+// three days, IsOffline will scan it, so the caller should treat this call as
+// blocking. If the host is not present in the HostDB, IsOffline returns false.
 func (hdb *HostDB) IsOffline(addr modules.NetAddress) bool {
+	// lookup entry
 	hdb.mu.RLock()
-	defer hdb.mu.RUnlock()
-
-	if _, ok := hdb.activeHosts[addr]; ok {
+	var lastSeen, lastScanned time.Time
+	entry, ok := hdb.allHosts[addr]
+	if ok {
+		lastSeen, lastScanned = entry.LastSeen, entry.LastScanned
+	}
+	hdb.mu.RUnlock()
+	if !ok {
 		return false
 	}
-	if h, ok := hdb.allHosts[addr]; ok {
-		return !h.Online
+
+	if time.Since(lastScanned) > uptimeThreshold {
```

Comment:
blocking isn't a big deal since the contractor is calling it in a background thread. And it's probably good because it means contract hosts will be scanned more frequently than other hosts in the hostdb.

regardless, I don't think two data points are really enough here. We want something more like "was successfully scanned 8/10 times over the last 3 days." Otherwise the false positive rate is just too high.